web/html/src/manager/systems/details/mgr-server-info.tsx

```diff
@@ -0,0 +1,124 @@
+import * as React from "react";
+
+import { TopPanel } from "components/panels/TopPanel";
+import { AsyncButton } from "components/buttons";
+
+import { Messages } from "components/messages";
+import { Utils as MessagesUtils } from "components/messages";
+
+import * as Systems from "components/systems/details";
+
+import Network from "utils/network";
+
+// See java/code/src/com/suse/manager/webui/templates/systems/mgr-server.jade
+type Props = {
+  serverId: string;
+  name: string;
+  version: string;
+  reportDbName: string;
+  reportDbHost: string;
+  reportDbPort: number;
+  reportDbUser: string;
+  reportDbLastSynced?: string;
+  isAdmin: boolean;
+  /** Locale of the help links */
+  docsLocale: string;
+};
+
+type State = {
+  messages: any[];
+};
+
+const msgMap = {
+  "Not a system id": t("Not a system id"),
+  "Unknown System": t("Unknown System"),
+  "System is not a Mgr Server": t("System is not a peripheral server"),
+};
+
+class MgrServer extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      messages: [],
+    };
+  }
+
+  handlePwReset() {
+    return Network.post(`/rhn/manager/api/systems/${this.props.serverId}/mgr-server-reportdb-newpw`, { })
+      .then((data) => {
+        if (data.success) {
+          this.setState({
+            messages: MessagesUtils.success(<a href={"/rhn/systems/details/history/History.do?sid=" + this.props.serverId}>
+                                            {t("New password generated and action for peripheral server scheduled")}</a>),
+          });
+        } else {
+          this.setState({
+            messages: MessagesUtils.error(data.messages.map((m) => msgMap[m])),
+          });
+        }
+      })
+      .catch((jqXHR) => {
+        this.handleResponseError(jqXHR);
+      });
+  };
+
+  handleResponseError = (jqXHR, arg = "") => {
+    this.setState({
+      messages: Network.responseErrorMessage(jqXHR, (status, msg) => (msgMap[msg] ? t(msgMap[msg], arg) : msg)),
+    });
+  };
+
+  render() {
+    return (
+      <div>
+        {this.state.messages ? <Messages items={this.state.messages} /> : null}
+        <TopPanel
+         title={t("Peripheral Server")}
+         icon={"spacewalk-icon-suse-manager"}
+         helpUrl={"reference/systems/systems-list.html"}
+        >
+          <div className="panel panel-default">
+            <div className="panel-body">
+              <React.Fragment>
```

Comment:
This fragment can safely be removed.  

<hr>

A slightly longer explanation if you're interested why:  

Fragments are necessary when you want to a-la return multiple root nodes from a function without a wrapper node, or if you want to conditionally render multiple nodes without a wrapper node etc. In a regular DOM tree like this, they are no-op and are safe to remove.  

Essentially, fragments are a crutch for React, which otherwise has a hard time figuring out what elements to replace and what to keep in situations where the number of returned children changes:  

```
render() {
  return (
    <React.Fragment>
      {(Math.random() > 0.5) ? (
        <div />
      ) : (
        <div />
        <div />
      )}
    </React.Fragment>
  );
}
```

If there's no multiple roots nor any conditional rendering like this involved, fragments are not necessary.

Reply:
removed - I do not need to understand why I need it :-)